app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,126 @@
+import android.content.pm.PackageManager;
+import android.support.annotation.NonNull;
+import android.widget.ImageButton;
+
+import com.otaliastudios.cameraview.CameraListener;
+import com.otaliastudios.cameraview.CameraView;
+import com.otaliastudios.cameraview.Gesture;
+import com.otaliastudios.cameraview.GestureAction;
+
+/**
+ * Activity for camera preview and taking photos
+ */
+public class CameraActivity extends AppCompatActivity {
+
+    //variable used to interface with the CameraView library
+    private CameraView camera;
+
+    //variable (with default value false) set to true if a picture is captured
+    //only when the picture is sent to the PicturePreviewActivity the variable is set to false again
+    private boolean isPictureCaptured = false;
+
+    /**
+     * onCreate method of the Android Activity Lifecycle
+     * @param savedInstanceState the bundle of the last instance state saved
+     * @author Bedin Elia
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_camera);
+        camera = findViewById(R.id.viewCamera);
+
+        //interface method for automatic onResume, onPause, onDestroy
+        camera.setLifecycleOwner(this);
+
+        //Pinch to zoom support
+        camera.mapGesture(Gesture.PINCH, GestureAction.ZOOM);
+        //Tap to focus support
+        camera.mapGesture(Gesture.TAP, GestureAction.FOCUS_WITH_MARKER);
+        //Long tap to capture support (currently disabled, should be .CAPTURE)
+        camera.mapGesture(Gesture.LONG_TAP, GestureAction.NONE);
+
+        camera.addCameraListener(new CameraListener() {
+            /**
+             * Listening for photo taken by the camera
+             * @param picture the taken photo in array form (stream of byte)
+             * @author Bedin Elia
+             */
+            public void onPictureTaken(byte[] picture) { onCapture(picture); }
+        });
+
+        ImageButton buttonPhoto = findViewById(R.id.capturePhoto);
+        buttonPhoto.setOnClickListener(new View.OnClickListener() {
+            /**
+             * Listening for button click, then call capturePhoto() to take a photo
+             * @param view the layout components
+             * @author Bedin Elia
+             */
+            @Override
+            public void onClick(View view) {
+                capturePhoto();
+            }
+        });
+    }
+
+    /**
+     * Method for taking a photo
+     * @author Bedin Elia
+     */
+    private void capturePhoto() {
+        if (isPictureCaptured) return;
+        isPictureCaptured = true;
+        camera.capturePicture();
+    }
+
+    /**
+     * Method for sending the taken photo in array form to the PicturePreviewActivity
+     * @param picture the taken photo in array form (stream of byte)
+     * @author Bedin Elia
+     */
+    private void onCapture(byte[] picture) {
+        isPictureCaptured = false;
```

Comment:
Could be a good idea making the buttonPhoto a private shared variable for this class, removing isPictureCaptured and the relative statements, putting on this line (99) `buttonPhoto.setEnabled(true);` and putting in the onClick method of the line 77 `buttonPhoto.setEnabled(false)`?

Reply:
This is a proposal to avoid a boolean synchronization if it could be useful.
When the button is pressed the button is disabled and it is enabled when the processing is ended.

Reply:
Sorry if it took me a long time to apply and test the proposed changes.
They are now applied in the new commit (https://github.com/PrandiniUniPD/OCRCamera/commit/753e923d2dc820a0883776c8fc36e5c847c8a4dc)
Thank you very much for you help on making the code better!